### PR TITLE
Update to otel-rust v0.23

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #
Design discussion issue (if applicable) #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-user-events-metrics/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Changed
+
+- Bump opentelemetry version and opentelemetry_sdk versions to 0.23,
+  opentelemetry-proto version to 0.5.
```
Comment (cijothomas) [inline opentelemetry-user-events-metrics/CHANGELOG.md:8]: eventheader

**Review by cijothomas — APPROVED** [review]

**Comment by SimenB** [comment]
Will this be published? 🙏

**Comment by hdost** [comment]
@lalitb do you want to take care of it or do you want me to?

**Comment by lalitb** [comment]
@hdost 
I and @cijothomas will handle the `opentelemetry-user-events-*` and `opentelemetry-etw-*` crates. If you could help with rest, that would be great. Otherwise, I can try to find some time next week to work on them. I’m not very familiar with how to validate the `dynatrace`, `slackdriver`, and `datadog` crates before publishing, so we’ll need to look into that as well.

**Pull request merged**